R/stan_polr.R

```diff
@@ -141,6 +141,11 @@ stan_polr <- function(formula, data, weights, ..., subset,
                       do_residuals = NULL) {
 
   data <- validate_data(data, if_missing = environment(formula))
+  is_char <- which(sapply(data, is.character))
+  for (j in is_char) {
+    data[[j]] <- as.factor(data[[j]])
+  }
+  
```

Comment:
Need to convert any character predictors to factors to avoid a potential error in model.matrix() 

```
Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : 
  contrasts can be applied only to factors with 2 or more levels
```

which happens when `newdata` is specified for reloo or kfold and a character predictor that had multiple unique values only has 1 in `newdata`. Converting to factor immediately when fitting initial model preserves the other levels even if they're not present in `newdata`.

Reply:
OK, whatever